Index: sonata/pluginsystem.py
===================================================================
--- sonata/pluginsystem.py	(revision 1238)
+++ sonata/pluginsystem.py	(working copy)
@@ -21,6 +21,7 @@
 		self.name = name
 		self._info = info
 		self._load = load
+		self.app = self
 		# obligatory plugin info:
 		format_value = info.get('plugin', 'plugin_format')
 		self.plugin_format = tuple(map(int, format_value.split(',')))
@@ -181,6 +182,7 @@
 		# XXX load from a .py file - no .pyc etc.
 		__import__('sonata.plugins', {}, {}, [name], 0)
 		plugin = getattr(sonata.plugins, name)
+		plugin.APP = self.app
 		return plugin
 
 pluginsystem = PluginSystem()
Index: sonata/main.py
===================================================================
--- sonata/main.py	(revision 1238)
+++ sonata/main.py	(working copy)
@@ -785,6 +785,7 @@
 		gc.disable()
 		
 		gobject.idle_add(self.header_save_column_widths)
+		pluginsystem.app = self
 
 		pluginsystem.notify_of('tabs', 
 				       self.on_enable_tab, 

